firestore/add_data.rb

```diff
@@ -0,0 +1,190 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+require "date"
+
+def set_document project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_set_document]
+  city_ref = firestore.doc "cities/LA"
+
+  data = {
+    name: "Los Angeles",
+    state: "CA",
+    country: "USA"
+  }
+
+  city_ref.set(data)
+  # [END fs_set_document]
+  puts "Set data for the LA document in the cities collection."
+end
+
+def update_create_if_missing project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
```

Comment:
Remove parentheses and add space.

project_id:  "value"    ---> this is a 'keyword parameters' and is recommended. This makes it a hash so the order parameters are included does not matter.

:project_id => "value"        ---> this is a 'symbol' -> basically does same thing but is optimization. Use case for this is :"stringliteral" => "value". Discouraged because of GitHub style guide.

"value"    ---> this is a 'positional parameter' (used with def update_create_if_missing project_id )

Frank recommends using keyword parameters.